src/battle/battle_script_commands.c

```diff
@@ -1186,6 +1186,76 @@ BOOL btl_scr_cmd_54_ohko_move_handle(void *bw, struct BattleStruct *sp)
     return FALSE;
 }
 
+BOOL btl_scr_cmd_7c_beat_up_damage_calc(void *bw, struct BattleStruct *sp)
+{
+    int species, form;
+    s32 newBaseDamage;
+    struct BattlePokemon *mon;
+    
+    IncrementBattleScriptPtr(sp, 1);
+
+    int partyCount = GetPartySize(bw, sp->attack_client);
+
+    if (sp->multi_hit_count_temp == 0) {
+
+        sp->multi_hit_count_temp = 2;
+        sp->loop_hit_check = 0xFD;
+        sp->beat_up_count = 0;
+        mon = GetPartyMon(bw, sp->attack_client, sp->beat_up_count); 
+
+        while(!(sp->beat_up_count != sp->sel_mons_no[sp->attack_client] ||
+                (GetMonData(mon, ID_PARA_hp, 0) && 
+                GetMonData(mon, ID_PARA_monsno_egg, 0) && 
+                GetMonData(mon, ID_PARA_monsno_egg, 0) != 494 && 
+                GetMonData(mon, ID_PARA_condition, 0) == 0)))
```

Comment:
that second side of the `||` is checking for if hp is not zero, if the species is not zero, if the species is not an egg, and if the nonvolatile status conditions are nonzero

Reply:
Got the logic of this check confused, corrected to
```
while(sp->beat_up_count != sp->sel_mons_no[sp->attack_client] &&
                (GetMonData(mon, ID_PARA_hp, 0) == 0 || 
                GetMonData(mon, ID_PARA_monsno_egg, 0) == 0 || 
                GetMonData(mon, ID_PARA_monsno_egg, 0) == 494 || 
                GetMonData(mon, ID_PARA_condition, 0) != 0))
```